benchmark/scripts/find_heap_bound.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# See LICENSE for license details.
+
+import re
+import argparse
+from typing import NamedTuple
+import logging
+
+from monitor_job import monitor_job, JobFailedError
+
+BaseHeapSize = NamedTuple('JavaHeapSize', [('value', int), ('suffix', str)])
+class HeapSize(BaseHeapSize):
+    K_FACTOR = 1024
+    M_FACTOR = 1024*1024
+    G_FACTOR = 1024*1024*1024
+
+    def toBytes(self) -> int:
+        if self.suffix == "":
+            factor = 1
+        elif self.suffix == "K":
+            factor = self.K_FACTOR
+        elif self.suffix == "M":
+            factor = self.M_FACTOR
+        elif self.suffix == "G":
+            factor = self.G_FACTOR
+        return self.value * factor
```

Comment:
Consider a dictionary for structured information. Ditto for other places in this file.

```suggestion
        { "":  1,
          "K": self.K_FACTOR,
          "M": self.M_FACTOR,
          "G": self.G_FACTOR }[self.suffix] * self.value
```